astropy/io/fits/hdu/image.py

```diff
@@ -577,7 +577,10 @@ def _scale_internal(
             self._header["BLANK"] = blank
 
         if self.data.dtype.type != _type:
-            self.data = np.array(np.around(self.data), dtype=_type)
+            if issubclass(_type, np.floating):
+                self.data = np.array(self.data, dtype=_type)
+            else:
+                self.data = np.array(np.around(self.data), dtype=_type)
```

Comment:
I couldn't help but observe that this does a potentially wasteful copy, which could be avoided as
```python
self.data = np.around(self.data).astype(_type, copy=COPY_IF_NEEDED)
```

Reply:
Oh I see. I missed that copying wasn't avoidable in this case.